### PR TITLE
Updates (Look at description)

### PR DESCRIPTION
**Pull request by SoVinhou**
This update includes:
- Ability for user to now create account
- Ability for user to now Sign In with their email accounts
- Ability for user to sign in using their google account (using google oauth)
- Add user data to database when they create account into Firebase Database
- Updated Designs
- Reset Password Page and ability for user to reset password through their email address

### CODE REVIEW

**Review by Vortrox — CHANGES_REQUESTED** [review]
We need to merge PR #28 first before merging this

File: Frontend/CourseFlow_Login_Form/Firebase Account/Account.txt
Comment (cmarktopher) [inline Frontend/CourseFlow_Login_Form/Firebase Account/Account.txt:1]: Might not be a good idea to include these.

File: Frontend/CourseFlow_Login_Form/src/environments/environment.prod.ts
Comment (cmarktopher) [inline Frontend/CourseFlow_Login_Form/src/environments/environment.prod.ts:1]: Same with the account details, might not want to include these here either.

File: Frontend/CourseFlow_Login_Form/src/environments/environment.ts
Comment (cmarktopher) [inline Frontend/CourseFlow_Login_Form/src/environments/environment.ts:1]: Same with the account details, might not want to include these here either.

**Comment by SoVinhou** [comment]
I have made the requested changes:
- resetted courseflow firebase password
- hide API keys (Only students with access to account which will be provided privately) will be able to run application

**Review by cmarktopher — APPROVED** [review]
Review of PR:

- Accounts and keys were updated and information removed from recent commit.
- Tested features locally.
- Email/Password based sign up works with email verification also working.
- Google OAuth is also working.
- Reset password with reset email link also working correctly. New password works with login.

**Comment by cmarktopher** [comment]
PR #28 seems to have been merged already.

**Comment by SoVinhou** [comment]
Overall Video Demonstration of the pull request:
https://youtu.be/UXUS8YocjVo

**Pull request merged**